### PR TITLE
[Modular] [Bounty] Adds the ability to 'army crawl'

### PR DESCRIPTION
**Pull request by Odairu**
## About The Pull Request
This PR adds the ability to lower yourself even further onto the ground than normal resting, allowing you to crawl under tables/railings and the like
Being prone in this manner will prevent you from attacking, pulling, and getting back up onto your feet until you've gotten back up from the prone state, while bullets and lasers will go over you, you are pretty helpless in this state to prevent abuse cases

## Why It's Good For The Game
This is mostly for under the table shenanigans (ERP purposes) elsewise I can't really think of any other use case.

## Proof Of Testing
https://youtu.be/j1WBvLybH50

## Changelog

:cl:
add: added the ability to army crawl below the table layer
/:cl:

### CODE REVIEW

**Review by projectkepler-ru — APPROVED** [review]
Code looks fine + bounty

**Review by Majkl-J — CHANGES_REQUESTED** [review]
The bounty specifically states this should be made into a module for any (presumably living) mobs, yet the component/element is limited to carbons. Also this is better off to do upstream so we can add the functionality to borers too and not have 2 different cases of the same behaviour defined differently in the code

File: modular_zubbers/code/datums/components/prone_mob.dm [hunk 1]
```diff
@@ -0,0 +1,38 @@
+/datum/component/prone_mob
+	var/mob/living/pronemob
+
+/datum/component/prone_mob/Initialize(...)
+	. = ..()
+	if(!iscarbon(parent))
+		return COMPONENT_INCOMPATIBLE
```
[the remark this answers is not in the document]
Comment (Odairu) [reply modular_zubbers/code/datums/components/prone_mob.dm]: Done

File: modular_zubbers/code/modules/mob/mob_movement.dm [hunk 2]
```diff
@@ -0,0 +1,18 @@
+/mob/verb/army_crawl()
+	set name = "Army Crawl"
+	set category = "IC"
+
+	var/mob/living/crawler = src
+	if(!crawler.resting)
+		return balloon_alert(src, "you must be laying down to army crawl.")
```
[the remark this answers is not in the document]
Comment (Odairu) [reply modular_zubbers/code/modules/mob/mob_movement.dm]: done

**Comment by YakumoChen** [comment]
I don't see the code regarding layering with tables and the like, or any sanity checks for if the crawler is under a table when they stop crawling (so they don't presumably phase through the table)

Army crawling should inflict a more severe crawl slowdown to stem abuse from the fact they dodge projectiles (people will just abuse this to be annoying as they are unclickable or dance between flaps or stairs

This is also going to make snail drifting even more annoying as well

**Review by projectkepler-ru — CHANGES_REQUESTED** [review]
On second review has this actually been tested on any other things to make sure they do work and have the intended interaction?

**Comment by Odairu** [comment]
> I don't see the code regarding layering with tables and the like, or any sanity checks for if the crawler is under a table when they stop crawling (so they don't presumably phase through the table).
> 
> Army crawling should inflict a more severe crawl slowdown to stem abuse from the fact they dodge projectiles (people will just abuse this to be annoying as they are unclickable or dance between flaps or stairs. The enhanced click protection is going to be abused by people trying to spite security for sure.
> 
> This is also going to make snail drifting even more annoying as well
> 
> I am sure people are going to find just a lot of exploits in general this doesn't cover, as one would expect.

They cant attack, pull, use ranged weapons, or throw objects in this state, they also cannot stand back up, and the only way to exit it is with the 3 second do_after requiring them to stand still for that time. 
They dont need more slowdown because if someone enters this state you just melee them and they literally cannot do anything back to you

**Comment by YakumoChen** [comment]
How are you supposed to melee them when their entire sprite is hidden under a table

**Comment by Majkl-J** [comment]
> Can you address the other legitimate concerns

Altclick table - click sprite.
The same counter works on borers, and nobody ever had concerns over that even though borers can still fire off abilities from under tables

**Comment by Odairu** [comment]
This got PRed up on SR

**Comment by YakumoChen** [comment]
> Altclick table - click sprite.
The same counter works on borers, and nobody ever had concerns over that even though borers can still fire off abilities from under tables

Alt clicking and loading up the menu is not an instant process, you will not be able to do this on someone actively moving around

**Pull request closed without merging**